### PR TITLE
[backup] fix that proof was asked for the wrong key

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

Fix a bug.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

test updated
## Related PRs

blame #3890 https://github.com/libra/libra/commit/41f065bb5a639b8c851d87301ba7eacd7c39892d

### CODE REVIEW

File: storage/backup/backup-cli/src/backup.rs [hunk 1]
```diff
@@ -77,8 +77,9 @@ pub async fn backup_account_state(
                 .write_new_file(stream::once(async move { chunk }))
```
Comment (wqfish) [inline storage/backup/backup-cli/src/backup.rs:77]: what is the "4" on line 74?

File: storage/backup/backup-cli/src/tests.rs [hunk 2]
```diff
@@ -54,12 +54,7 @@ fn end_to_end() {
     let client = BackupServiceClient::new(config.storage.backup_service_port);
     let (version, state_root_hash) = rt.block_on(client.get_latest_state_root()).unwrap();
     let handles = rt
-        .block_on(backup_account_state(
-            &client,
-            version,
-            &adaptor,
-            1024 * 1024,
-        ))
+        .block_on(backup_account_state(&client, version, &adaptor, 500))
```
Comment (wqfish) [inline storage/backup/backup-cli/src/tests.rs:57]: is this because 1024^2 is too large and the test generates only one chunk?

Comment (msmouse) [reply storage/backup/backup-cli/src/tests.rs:57]: yes

File: storage/backup/backup-cli/src/backup.rs (hunk 1 above)
Comment (msmouse) [reply storage/backup/backup-cli/src/backup.rs:78]: it's the u32 indicating len of the record. (made explicit in new commit)

**Pull request closed without merging**